closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -2354,14 +2362,138 @@ Java_jdk_crypto_jniprovider_NativeCrypto_ECCreatePrivateKey
 
     (*env)->ReleasePrimitiveArrayCritical(env, s, nativeS, JNI_ABORT);
 
+    if (NULL != sBN) {
+        (*OSSL_BN_free)(sBN);
+    }
+
     if (0 == ret) {
         return -1;
     }
 
     return 0;
 }
 
-/* Encode an EC Elliptic Curve over a Prime Field
+/* Encode an EC Elliptic Curve over a Prime Field */
+EC_KEY* ECEncodeGFp(BIGNUM *aBN,
+                    BIGNUM *bBN,
+                    BIGNUM *pBN,
+                    BIGNUM *xBN,
+                    BIGNUM *yBN,
+                    BIGNUM *nBN,
+                    BIGNUM *hBN)
+{
+    EC_KEY *key = NULL;
+    EC_GROUP *group = NULL;
+    EC_POINT *generator = NULL;
+    BN_CTX *ctx = NULL;
+    int ret = 0;
+
+    ctx = (*OSSL_BN_CTX_new)();
+    if (NULL == ctx) {
+        goto cleanup;
+    }
+
+    group = (*OSSL_EC_GROUP_new_curve_GFp)(pBN, aBN, bBN, ctx);
+    if (NULL == group)
+        goto cleanup;
```

Comment:
Please don't omit braces (please correct throughout):
```c
    if (NULL == group) {
        goto cleanup;
    }
```

Reply:
This update should be done throughout all the code that I changed.